hyperon_das_atomdb/adapters/server_db.py

```diff
@@ -0,0 +1,277 @@
+import json
+from typing import Any, Dict, List, Optional, Tuple
+
+from requests import Session, exceptions
+
+from hyperon_das_atomdb.exceptions import (
+    ConnectionServerException,
+    NodeDoesNotExistException,
+)
+from hyperon_das_atomdb.i_database import IAtomDB
+from hyperon_das_atomdb.utils.decorators import retry
+
+
+class ServerDB(IAtomDB):
+    """A concrete implementation using servers databases.
+    AwsLambda and OpenFaas"""
+
+    def __repr__(self) -> str:
+        return "<Atom database Server>"  # pragma no cover
+
+    def __init__(
+        self,
+        database_name: str = 'das',
+        ip_address: str = None,
+        port: Optional[str] = None,
+    ) -> None:
+        if not ip_address:
+            raise ConnectionServerException(
+                message='You must send ip_address parameter'
+            )
+        self.database_name = database_name
+        self.ip = ip_address
+        self.port = port if port else '8080'
+        self.openfaas_uri = f'http://{self.ip}:{self.port}/function/atomdb'
+        self.aws_lambda_uri = f'http://{self.ip}/prod/atomdb'
+        self.session = Session()
+        self._connect_server()
+
+    @retry(attempts=5, timeout_seconds=120)
+    def _connect_server(self) -> str | None:
```

Comment:
May I suggest to change this design slightly.

In the constructor you should do

```
self.url = self._connect_server()
```
`openfaas_uri` and `aws_lambda_uri` should be local variables built inside `_connect_server()`.

Reply:
Why do you think this is better, do you think it's more explicit for anyone reading the code?
I'm going to ask because for me it's the same thing

Reply:
`openfaas_uri` and `aws_lambda_uri` are not used anywhere else so I don't see the point of having them as fields in the object. I'm not sure about `url` either. If it's also not used it should be cut as well and the method should return `session`

Reply:
ok. I got it